packages/gestalt/src/boxTypes.js

```diff
@@ -62,6 +62,32 @@ export type Display =
 export type Direction = 'row' | 'column';
 export type Flex = 'grow' | 'shrink' | 'none';
 export type Fit = boolean;
+export type Gap =
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
```

Comment:
@dangerismycat do we need to go all the way to `24` here? All of these will require extra generated CSS.

Looks like previously we only went to `12`

![image](https://user-images.githubusercontent.com/127199/98746871-33a73100-236b-11eb-86f0-52153d2d5896.png)

Reply:
The current implementation applies the gap value to each side, so 1 boint actually means 8px between elements, with a max of 96px between elements. However, this wasn't very flexible, since it was impossible to set a gap of odd numbers of actual boints.

The new implementation splits the gap value between each adjoining pair of elements, allowing for a single boint gap. However, this means the largest gap is only 48px, which is pretty small. I bumped this up to 24 to allow for a larger total gap, similar to what we can currently achieve. (This is also necessary to be a non-breaking change, though I don't know if we have any `gap={12}` usages.)

Happy to change it back if we want to be more restrictive instead, but I fear that will lead to lower adoption.

Reply:
@dangerismycat Thanks for explaining, we should also make the divide by 2 change in the codemod. I know in certain places we will pass `gap` dynamically but then we can throw an error to manually check.

Since we only have one place in the pinterest codebase which uses a legacy `gap={12}`, we should stick to only allowing up to 12 (since that's what we allow on other components as well). I know that limits the usage but we can always add it in if there is a big demand.

Reply:
I discussed with @ashleyseto and we agree, so I'll change this back to 12. We can up it to 24 in the future if needed.